src/volttron/server/aip.py

```diff
@@ -681,8 +667,17 @@ def install_agent_source(self, agent: str, force: bool = False, pre_release: boo
             # In such case, we should be able to revert to current version - so explicitly find current version and
             # see if that can be installed from pypi because local wheel might not be there anymore.
             try:
-                cmd_dry_run = ["poetry", "--directory", self._server_opts.poetry_project_path.as_posix(), "add",
-                               f"{agent_name}=={current_version}", "--dry-run"]
+                if agent.endswith(".whl"):
```

Comment:
@craig8 why is there a dry run with new agent whl? Here all we are trying to do is see if we can restore the currently installed version of agent in case the install of the pass "agent" (.whl or otherwise) fails

Reply:
So this is part of the issue that I was talking about.  I will be dealing with this in a later PR.